x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/use_overall_stats.ts

```diff
@@ -190,14 +213,29 @@ export function useOverallStats<TParams extends OverallStatsSearchStrategyParams
 
       const sub = rateLimitingForkJoin<
         AggregatableFieldOverallStats | NonAggregatableFieldOverallStats | undefined
-      >([...aggregatableOverallStatsObs, ...nonAggregatableFieldsObs], MAX_CONCURRENT_REQUESTS);
+      >(
+        [
+          nonAggregatableFieldsExamplesObs,
+          ...aggregatableOverallStatsObs,
+          ...nonAggregatableFieldsObs,
+        ],
+        MAX_CONCURRENT_REQUESTS
+      );
 
       searchSubscription$.current = sub.subscribe({
         next: (value) => {
           const aggregatableOverallStatsResp: AggregatableFieldOverallStats[] = [];
           const nonAggregatableOverallStatsResp: NonAggregatableFieldOverallStats[] = [];
 
+          let sampledNonAggregatableFieldsExamples: Array<{ [key: string]: string }> | undefined;
           value.forEach((resp, idx) => {
+            if (idx === 0 && isNonAggregatableSampledDocs(resp)) {
+              const docs = resp.rawResponse.hits.hits.map((d) =>
+                getProcessedFields(d.fields ?? {})
```

Comment:
rather than calling `getProcessedFields` with an empty object, it might be neater to make the check beforehand.
e.g.
```ts
d.fields ? getProcessedFields(d.fields) : {}
```